great_expectations/checkpoint/checkpoint.py

```diff
@@ -256,23 +263,32 @@ def _sort_actions(self) -> List[CheckpointAction]:
     def save(self) -> None:
         from great_expectations.data_context.data_context.context_factory import project_manager
 
-        store = project_manager.get_checkpoints_store()
-        key = store.get_key(name=self.name, id=self.id)
-
-        store.update(key=key, value=self)
-
-    def _add_to_store(self) -> None:
-        """This is used to persist a checkpoint before we run it.
-
-        We need to persist a checkpoint before it can be run. If user calls runs but hasn't
-        persisted it we add it for them.
-        """
-        from great_expectations.data_context.data_context.context_factory import project_manager
+        for validation in self.validation_definitions:
```

Comment:
If we want to be careful about rollbacks and not leaving us in an inbetween state, we'd need to keep track of current state so we can restore if we hit an exception part way through.